pkg/services/ngalert/store/alert_rule.go

```diff
@@ -33,6 +34,11 @@ const AlertRuleMaxTitleLength = 190
 // AlertRuleMaxRuleGroupNameLength is the maximum length of the alert rule group name
 const AlertRuleMaxRuleGroupNameLength = 190
 
+// RuleVersionRecordLimits defines the limit of how many alert rule versions
+// should be stored in the database for each alert_rule in an organization including the current one.
+// Has to be > 0
+const RuleVersionRecordLimits = 100
```

Comment:
100 is a fairly large amount of alert rule versions and seems like a good conservative number. That being said, I still wonder if this should be made opt-in for now. Preferably by making the limit configurable and default to 0 (unlimited).

Most will likely see the cleanup as a plus but I can imagine it being unexpected to some users who might want to keep more version history to suddenly have them deleted in a minor version update. We can always make it opt-out later.

I can point you to some example code to work from or I can add the configuration code myself if you prefer not to.

Reply:
I took a stab at adding it to the configuration. Let me know if there is a better place to add the configuration.